### DISCUSSION TITLE
Customising breakpoints conditions in sprinkles as a consumer

### DISCUSSION THREAD

**Discussion by imbhargav5** [vanilla-extract-css/vanilla-extract#800] [Q&A]
Hi VE community. I need help with a query related to extending breakpoints as an end user. 

I am building a UI Kit like so

```javascript
const sprinkles = createSprinkles({ 
    conditions: {
      mobile: {},
      tablet: { '@media': 'screen and (min-width: ' + breakpoints.$md + ')' },
      desktop: { '@media': 'screen and (min-width: ' + breakpoints.$lg + ')' },
  },
})
```

I would like the consumer of the UI kit to be able to modify the breakpoints as they use the UI Kit. Unfortunately, media queries can't be used with CSS variables which means I can't really do 

```javascript
const sprinkles = createSprinkles({ 
    conditions: {
      mobile: {},
      tablet: { '@media': 'screen and (min-width: ' + vars.breakpoints.$md + ')' },
      desktop: { '@media': 'screen and (min-width: ' + vars.breakpoints.$lg + ')' },
  },
})
```

If this were possible, I could use `theme contracts` to allow an end user to create a theme out of the `vars`.  But since this is not possible, I was wondering what other options we have to allow users to modify the breakpoints from my UI kit. 

Thanks!

**Comment by jarrodpayne** [comment]
@imbhargav5 Did you ever find a solution for this?

**Reply by imbhargav5** [reply]
@paynecodes Unfortunately no. I gave up.
